sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala

```diff
@@ -46,6 +46,20 @@ trait CheckAnalysis extends PredicateHelper {
     }).length > 1
   }
 
+  private def checkLimitClause(limitExpr: Expression): Unit = {
+    if (!limitExpr.foldable) {
```

Comment:
Are you saying we should change the SQL parser? You know, if so, we also need to change the `TABLESAMPLE n ROWS`.

FYI, even Hive does not support foldable expressions.
```
hive> select * from t1 limit 1 + 1;
FAILED: ParseException line 1:25 missing EOF at '+' near '1'
hive> select * from t1 limit (1+1);
FAILED: ParseException line 1:23 extraneous input '(' expecting Number near '1'
```
I found that Impala supports it. 
http://www.cloudera.com/documentation/archive/impala/2-x/2-1-x/topics/impala_limit.html